rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -248,7 +248,21 @@ void SqliteStorage::write_locked(
             "' has not been created yet! Call 'create_topic' first.");
   }
 
-  write_statement_->bind(message->time_stamp, topic_entry->second, message->serialized_data);
+  try {
+    write_statement_->bind(message->time_stamp, topic_entry->second, message->serialized_data);
+  } catch (const SqliteException & exc) {
+    if (SQLITE_TOOBIG == exc.get_sqlite_return_code()) {
+      ROSBAG2_STORAGE_DEFAULT_PLUGINS_LOG_WARN_STREAM(
```

Comment:
I thought about using one of the throttle or once type macros, but I hesitated for this since it's simpler. But I'm open to suggestions on what to do and, if throttle, how often to throttle it.

Reply:
I agree with @gbiggs. And practically, it seems pretty unlikely that you'll have messages exceeding the maximum SQLite blob size that are also coming at a high rate, because you would need an unusually fast network connection for that to happen.